components/SideNav.tsx

```diff
@@ -4,21 +4,35 @@ import classNames from 'classnames';
 
 // Models
 import { Documentation } from '@models/Documentation.model';
+import { useEffect } from 'react';
+
+interface SideNavProps {
+    data: Documentation;
+    docPage?: string;
+    level?: number;
+    breadCrumb?: string[];
+    hide?: boolean;
+    search?: boolean;
+    topLevel?: boolean;
+}
 
 export const SideNav = ({
     data,
+    breadCrumb,
+    level = 0,
     hide = false,
     topLevel = true
-}: {
-    data: Documentation;
-    hide?: boolean;
-    topLevel?: boolean;
-}): JSX.Element => {
+}: SideNavProps): JSX.Element => {
     const [navItem, setNavItem] = useState({
         showSubList: null,
         active: null
     });
-
+    useEffect(() => {
+        setNavItem({
+            showSubList: breadCrumb[level],
+            active: breadCrumb[level]
+        });
+    }, [breadCrumb]);
```

Comment:
Why useEffect?

Reply:
We set the nav item when the `breadCrumb` changes which will change once.